src/components/drawer/bl-drawer.ts

```diff
@@ -0,0 +1,134 @@
+import {customElement, property} from "lit/decorators.js";
+import {CSSResultGroup, html, LitElement, TemplateResult} from "lit";
+import {event, EventDispatcher} from "../../utilities/event";
+import '../button/bl-button';
+import style from './bl-drawer.css';
+import {PropertyValues} from "lit";
+
+/**
+ * @tag bl-drawer
+ * @summary Baklava Drawer component
+ *
+ */
+
+@customElement('bl-drawer')
+export default class BlDrawer extends LitElement{
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  /**
+   * Sets drawer open-close status
+   */
+  @property({ type: Boolean, reflect: true })
+  open = false;
+
+  /**
+   *  Sets the drawer title
+   */
+  @property({type:String})
+  caption?: string;
+
+  /**
+   *  Sets the drawer embed url for iframe
+   */
+  @property({type:String})
+  embedUrl?: string;
+
+  /**
+   *  Sets the drawer external link
+   */
+  @property({type:String})
+  externalLink?: string;
+
+  /**
+   * Fires when the drawer is opened
+   */
+  @event('bl-drawer-open') private onOpen: EventDispatcher<object>;
+
+  /**
+   * Fires when the drawer is closed
+   */
+  @event('bl-drawer-close') private onClose: EventDispatcher<object>;
+
+  connectedCallback() {
+    super.connectedCallback();
+    window?.addEventListener('bl-drawer-open',(event) => {
+      if(event.target !== this)
+        this.closeDrawer();
+    });
+  }
+
+  updated(changedProperties: PropertyValues<this>) {
+    if (changedProperties.has('open')) {
+      this.toggleDialogHandler();
+    }
+  }
+
+  private toggleDialogHandler() {
+    if (this.open) {
+      this.onOpen({ isOpen: true });
+    } else {
+      this.onClose({ isOpen: false });
```

Comment:
Actually I do not know this dispatch logic literally.

Reply:
This object you send is the payload for the event. We attach useful information for the listeners of this event. So they can read it like:

```js
document.querySelector('bl-drawer').addEventListener('bl-drawer-open', (event) => {
   console.log(event.detail); // the object you send
});
```

It doesn't need to be set always, if there is no helpful internal data for this event. But for example, for the `bl-drawer-close` event, we can give information about if it's closed by user (by clicking close button) or automatically (because another drawer opened). This information can be helpful in some use cases.

Reply:
Then I can send information like this `this.onClick('Click event fired!');` ?

Reply:
To be honest, there is no need for this text. It doesn't make sense. But I just noticed currently we don't allow to fire events *without* payload. Maybe you can just send an empty string for now and I'll create another issue for allowing events without payloads. Better to put a `// FIXME: Allow events without payload` on top of that line.

Reply:
I created an issue for that: #328